src/main/java/gyro/google/compute/AbstractHealthCheckResource.java

```diff
@@ -0,0 +1,406 @@
+    }
+
+    /**
+     * The ssl Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckSslHealthCheck
+     */
+    @Updatable
+    @ConflictsWith({ "https-health-check", "http2-health-check", "http-health-check", "tcp-health-check" })
+    public HealthCheckSslHealthCheck getSslHealthCheck() {
+        return sslHealthCheck;
+    }
+
+    public void setSslHealthCheck(HealthCheckSslHealthCheck sslHealthCheck) {
+        this.sslHealthCheck = sslHealthCheck;
+    }
+
+    /**
+     * The tcp Health Check type.
+     *
+     * @subresource gyro.google.compute.HealthCheckTcpHealthCheck
+     */
+    @Updatable
+    @ConflictsWith({ "https-health-check", "http2-health-check", "ssl-health-check", "http-health-check" })
+    public HealthCheckTcpHealthCheck getTcpHealthCheck() {
+        return tcpHealthCheck;
+    }
+
+    public void setTcpHealthCheck(HealthCheckTcpHealthCheck tcpHealthCheck) {
+        this.tcpHealthCheck = tcpHealthCheck;
+    }
+
+    /**
+     * The time (in seconds) to wait before claiming failure.
+     */
+    @Updatable
+    public Integer getTimeoutSec() {
+        return timeoutSec;
+    }
+
+    public void setTimeoutSec(Integer timeoutSec) {
+        this.timeoutSec = timeoutSec;
+    }
+
+    /**
+     * The type of health check. Valid values are: ``TCP``, ``SSL``, ``HTTP``, ``HTTPS`` or ``HTTP2``.
+     */
+    public String getType() {
+        return type;
+    }
+
+    public void setType(String type) {
+        this.type = type;
+    }
+
+    /**
+     * The number of consecutive failures before marking unhealthy.
+     */
+    @Updatable
+    public Integer getUnhealthyThreshold() {
+        return unhealthyThreshold;
+    }
+
+    public void setUnhealthyThreshold(Integer unhealthyThreshold) {
+        this.unhealthyThreshold = unhealthyThreshold;
+    }
+
+    /**
+     * The number of consecutive successes before marking healthy.
+     */
+    @Updatable
+    public Integer getHealthyThreshold() {
+        return healthyThreshold;
+    }
+
+    public void setHealthyThreshold(Integer healthyThreshold) {
+        this.healthyThreshold = healthyThreshold;
+    }
+
+    /**
+     * The server-defined URL for the health check.
+     */
+    @Output
+    public String getSelfLink() {
+        return selfLink;
+    }
+
+    public void setSelfLink(String selfLink) {
+        this.selfLink = selfLink;
+    }
+
+    @Override
+    public void copyFrom(HealthCheck healthCheck) {
+        setName(healthCheck.getName());
+        setDescription(healthCheck.getDescription());
+        setCheckIntervalSec(healthCheck.getCheckIntervalSec());
+        setTimeoutSec(healthCheck.getTimeoutSec());
+        setUnhealthyThreshold(healthCheck.getUnhealthyThreshold());
+        setHealthyThreshold(healthCheck.getHealthyThreshold());
+        setSelfLink(healthCheck.getSelfLink());
+        setType(healthCheck.getType());
+
+        if (healthCheck.getHttpHealthCheck() != null) {
```

Comment:
Set the subresource to null. If it is present in the state and config but deleted from the console, gyro would not be able to pick it up.

```suggestion
        setHttpHealthCheck(null);
        if (healthCheck.getHttpHealthCheck() != null) {
```